src/ConnectedMode/Migration/XmlFileCleaner.cs

```diff
@@ -73,12 +73,35 @@ public string Clean(string content, LegacySettings legacySettings, CancellationT
             nodesToRemove.ForEach(node =>
             {
                 Debug.Assert(node.ParentNode != null);
-                node.ParentNode.RemoveChild(node);
+
+                // If the only child elements in the parent element are white spaces and the node the whole parent elemnt can be removed.
+                var currentNode = ShouldRemoveParentElement(node) ? node.ParentNode : node;
+
+                // This will ensure that there is no white space left behind awhen removing the actual currentNode.
+                if (currentNode.PreviousSibling.NodeType == XmlNodeType.Whitespace)
+                {
+                    currentNode.ParentNode.RemoveChild(currentNode.PreviousSibling);
+                }
+
+                currentNode.ParentNode.RemoveChild(currentNode);
             });
 
             return xmlDocumentHelper.SaveToString(document);
         }
 
+        private static bool ShouldRemoveParentElement(XmlNode node)
+        {
+            foreach (XmlNode child in node.ParentNode.ChildNodes)
```

Comment:
I wanted to do a cool linq expression here but it would not let me for some reason. ChildNodes is IEnumerable so it should have been possible but anyways, thats why we have a normal foreach loop now